apollo-backend/src/main/java/io/logz/apollo/database/DataSourceFactory.java

```diff
@@ -21,23 +21,21 @@ private DataSourceFactory() {}
     public static DataSource create(DatabaseConfiguration databaseConfiguration) {
         Properties poolProperties = new Properties();
         poolProperties.setProperty("username", databaseConfiguration.getUser());
-        poolProperties.setProperty("dataSourceClassName", com.mysql.jdbc.jdbc2.optional.MysqlDataSource.class.getName());
+        poolProperties.setProperty("dataSourceClassName", org.mariadb.jdbc.MariaDbDataSource.class.getName());
```

Comment:
This single line is the only compile time dependency on the driver, right?
Can we move the data source class name to the properties file? That way we break the dependency and we don't need any explicit driver in the project classpath.
This is actually the recommended way of doing things: you should be able to switch out the database implementation by switching the JDBC driver and restarting, no compile changes required.